### PR TITLE
Expressions: Add notification for Strict Mode behavior in Reduce component

### LINKED ISSUE #79315: InfluxDB: Show a notification when fill(null) and reduce expression with strict mode
When `Reduce strict mode` is used, `fill(null)` (Influxql function) will result in `NaN`. 
See docs https://grafana.com/docs/grafana/latest/panels-visualizations/query-transform-data/expression-queries/#sum

This is a confusion for the users who don't know the details in deep. We can display a notification when `Reduce with strict mode` is used and point to the documentation.

### PR DESCRIPTION
**Pull request by shubhankarunhale**
**What is this feature?**

This feature adds a notification to the Reduce component in Grafana when "Strict Mode" is selected. It informs users about the behavior of the fill(null) function in InfluxQL, which results in NaN.

**Why do we need this feature?**

Users unfamiliar with the behavior of "Strict Mode" may find the NaN result confusing when using fill(null). This notification provides clarity and directs users to the relevant documentation, improving the user experience.

**Who is this feature for?**

This feature is for Grafana users who use the Reduce component with "Strict Mode" enabled, especially those working with InfluxQL queries.

**Which issue(s) does this PR fix?**:
Fixes 

Fixes #79315 

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feaxture toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

GraphanaFix

### CODE REVIEW

File: public/app/features/expressions/components/Reduce.tsx [hunk 1]
```diff
@@ -66,8 +72,30 @@ export const Reduce = ({ labelWidth = 'auto', onChange, refIds, query }: Props)
     );
   };
 
+  const strictModeNotification = () => {
+    if (mode !== ReducerMode.Strict) {
+      return null;
+    }
+    return (
+      <Alert title={t('reduce.strictMode.title', 'Strict Mode Behaviour')} severity="info">
+        <Trans i18nKey="reduce.strictMode.description">
```
Comment (itsmylife) [inline public/app/features/expressions/components/Reduce.tsx:81]: You need to add the key in this file https://github.com/grafana/grafana/blob/main/public/locales/en-US/grafana.json
And then run `make i18n-extract` then commit the files.

**Review by itsmylife — CHANGES_REQUESTED** [review]
LGTM but i18n check is failing due to the key must be added to the translations file. Could you please add it too?

File: public/app/features/expressions/components/Reduce.tsx (hunk 1 above)
Comment (shubhankarunhale) [reply public/app/features/expressions/components/Reduce.tsx:81]: Added the translations keys, an incorrect rebase which I now reverted, made the grafana-pr-automation add extra tags to the issue. I don't think I have the permission to remove these. 
@itsmylife Can you please review?

File: public/app/features/expressions/components/Reduce.tsx [hunk 2]
```diff
@@ -66,8 +72,30 @@ export const Reduce = ({ labelWidth = 'auto', onChange, refIds, query }: Props)
     );
   };
 
+  const strictModeNotification = () => {
+    if (mode !== ReducerMode.Strict) {
+      return null;
+    }
+    return (
+      <Alert title={t('reduce.strictMode.title', 'Strict Mode Behaviour')} severity="info">
```
Comment (itsmylife) [inline public/app/features/expressions/components/Reduce.tsx:80]: This looks good. Thank you!

File: public/app/features/expressions/components/Reduce.tsx (hunk 1 above)
Comment (itsmylife) [inline public/app/features/expressions/components/Reduce.tsx:81]: Let's split this into two parts. 
First should hold only the text till the url, and you can put this in json file too. I'll put a comment into the json too. 

```
When <code>Reduce Strict mode</code> is used, the <code>fill(null)</code> function (InfluxQL) will result in{' '}<code>NaN</code>.
```

and the second part should only hold the text inside the `a` tag. With this way when it's translated we won't lose the link. 

```
See the documentation for more details.
```

**Review by itsmylife — APPROVED** [review]
🚀

**Pull request merged**